### PR TITLE
Using honorLabels: true removes the need for relabeling.

### PR DESCRIPTION
**Pull request by fedepaol**
Also, add a deploy-k8s command to test it on vanilla k8s.

The changes in vendor are the result of a make deps-update. Not sure I missed that before.

### CODE REVIEW

File: deployments-k8s/03_daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      app: network-metrics-daemon
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: network-metrics-daemon
+        component: network
+        type: infra
+        openshift.io/component: network
```
Comment (SchSeba) [inline deployments-k8s/03_daemonset.yaml:19]: Can we remove this label as it's unrelated to k8s deployment

File: deployments-k8s/03_daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/worker: ""
```
Comment (SchSeba) [inline deployments-k8s/03_daemonset.yaml:22]: why we can't deploy this daemon on all the nodes?

File: deployments-k8s/03_daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (SchSeba) [inline deployments-k8s/03_daemonset.yaml:1]: Can you please add a note in the documents that for the k8s deploy we don't use a TLS connection from Prometheus to the pod

File: Makefile [hunk 4]
```diff
@@ -35,3 +35,5 @@ image_push: ; $(info Building image...)
 deploy:
 	hack/deploy.sh
 
+deploy-k8s:
```
Comment (SchSeba) [inline Makefile:38]: Can you add a section for the kubernetes deployment in the README file?

File: deployments-k8s/03_daemonset.yaml (hunk 2 above)
Comment (fedepaol) [reply deployments-k8s/03_daemonset.yaml:22]: I guess we can..

File: README.md [hunk 5]
```diff
@@ -52,3 +52,7 @@ This is achieved by joining the metrics produced by this daemon with the existin
 ## Deploy
 
 Running `make deploy` will deploy the daemonset and set up the configuration to tie it to the Prometheus operator instance of an existing OpenShift 4+ cluster.
+
+## Deploy on k8s
+
+Running `make deploy` will deploy the daemonset and set up the configuration to tie it to the Prometheus operator instance of an existing kubernetes cluster where the prometheus operator was deployed. Please note that the k8s version exposes metrics over a plain http port.
```
Comment (SchSeba) [inline README.md:58]: should it be `make deploy-k8s` ?

**Pull request merged**